cmake/concepts_ts.cmake

```diff
@@ -0,0 +1,54 @@
+#
+#  Copyright Christopher Di Bella
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+include(FindPackageHandleStandardArgs)
+include(CheckCXXCompilerFlag)
+include(CMakePushCheckState)
+
+message(STATUS "Checking for concept support")
+
+cmake_push_check_state(RESET)
+   if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
+      set(CMAKE_REQUIRED_FLAGS -fconcepts)
+   else()
+      check_cxx_compiler_flag(-fconcepts-ts CLANG_CONCEPTS_SUPPORT)
+
+      if (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS "6.0.0" OR NOT ${CLANG_CONCEPTS_SUPPORT})
+         message(FATAL_ERROR "You require a copy of clang-concepts to build clang-concepts-ranges.")
+      endif()
+
+      set(CMAKE_REQUIRED_FLAGS -std=c++2a -Xclang -fconcepts-ts)
+   endif()
+
+   check_cxx_source_compiles("template<class> concept Auto = true; int main() {}" cjdb_feature_test_concepts)
+   set(CMAKE_REQUIRED_FLAGS "")
+cmake_pop_check_state()
+
+set(CJDB_DEFINE_CONCEPT_AS_CONCEPT_BOOL OFF)
```

Comment:
Define an imported target and do `target_compile_options(Concepts::Concepts INTERFACE ...)`